packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -118,6 +120,74 @@ export type DirFileNameSelection = {
   outputdir: string;
 };
 
+export class SelectFileName
+  implements ParametersGatherer<{ fileName: string }> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ fileName: string }>
+  > {
+    const fileNameInputOptions = <vscode.InputBoxOptions>{
+      prompt: nls.localize('parameter_gatherer_enter_file_name')
+    };
+    const fileName = await vscode.window.showInputBox(fileNameInputOptions);
+    return fileName
+      ? { type: 'CONTINUE', data: { fileName } }
+      : { type: 'CANCEL' };
+  }
+}
+
+export class SelectDirPath
+  implements ParametersGatherer<{ outputdir: string }> {
+  private explorerDir: string | undefined;
+  private globKeyWord: string | undefined;
+
+  public constructor(explorerDir?: { path: string }, globKeyWord?: string) {
```

Comment:
Changed this parameter gatherer to accept a keyword so it can prioritize the dirs based on the type (apex class, vf page, vf component, etc)